src/platform/Darwin/CHIPPlatformConfig.h

```diff
@@ -27,6 +27,12 @@
 
 #define CHIP_CONFIG_ABORT() abort()
 
+#include <os/trace_base.h> // for __dso_handle
+extern "C" int __cxa_atexit(void (*f)(void *), void * p, void * d);
+#define CHIP_CXA_ATEXIT(f, p) __cxa_atexit((f), (p), &__dso_handle)
```

Comment:
Perhaps, this macro could just be called `CHIP_ATEXIT`  as some platforms may not provide `__cxa_atexit`?

Reply:
Regular atexit doesn't have the `void *` parameter so it's harder to use for this. Platforms don't need to provide this unless they use the lazy option without no_destruct.

Not sure the name matters too much -- the semantics are expected to be like `__cxa_atexit` so I named it based on that.